From fd7210aba7d168865960d7c5884bb214b45bef00 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Wed, 2 Sep 2020 11:22:28 -0700
Subject: [PATCH 1550/1851] drm: xlnx: pl_disp: Release DMA when probe deferred

commit 76b759c35458d5ff55cb7c8d6ecc9dfa432efc26 from
https://github.com/Xilinx/linux-xlnx.git

Release the DMA channel when probe is deferred due to VTC driver not
being loaded.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_pl_disp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_pl_disp.c b/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
index 8f810729d01a..d8e7a9f22779 100644
--- a/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
+++ b/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
@@ -572,7 +572,8 @@ static int xlnx_pl_disp_probe(struct platform_device *pdev)
 		xlnx_pl_disp->vtc_bridge = of_xlnx_bridge_get(vtc_node);
 		if (!xlnx_pl_disp->vtc_bridge) {
 			dev_info(dev, "Didn't get vtc bridge instance\n");
-			return -EPROBE_DEFER;
+			ret = -EPROBE_DEFER;
+			goto err_dma;
 		}
 	} else {
 		dev_info(dev, "vtc bridge property not present\n");
-- 
2.31.1

